core/persistence-jpa/src/main/java/org/apache/syncope/core/persistence/jpa/dao/JPATaskDAO.java

```diff
@@ -436,7 +413,13 @@ public int count(
     @Transactional(rollbackFor = { Throwable.class })
     @Override
     public <T extends Task> T save(final T task) {
-        return entityManager().merge(task);
+        T merged = entityManager().merge(task);
+
+        // propagate the event only for Propagation tasks
+        if (merged instanceof PropagationTask) {
+            publisher.publishEvent(new TaskCreatedUpdatedEvent<>(this, merged, AuthContextUtils.getDomain()));
```

Comment:
Maybe `TaskCreatedUpdatedEvent` shall be renamed as `PropagationTaskCreatedUpdatedEvent`?

Reply:
Yes, I left the implementation in a way "open" also to other tasks. There are some cases in which would be useful also to have `NotificationTask` on ES.

Reply:
It does not make much sense to leave open about things that will never be invoked, as the only caller can realistically be `JPATaskDAO`, and a clear `if` is placed there around publisher invocation.